Error when running level command without UltimateStacker

### Expected behavior

No error

### Observed/Actual behavior

```
[15:13:15 INFO]: mrfloris issued server command: /skyblock level
[15:13:15 ERROR]: null
org.bukkit.command.CommandException: Unhandled exception executing 'skyblock level' in world.bentobox.bskyblock.BSkyBlock$1(skyblock)
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:165) ~[paper-api-1.20.4-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_20_R3.CraftServer.dispatchCommand(CraftServer.java:999) ~[paper-1.20.4.jar:git-Paper-488]
        at org.bukkit.craftbukkit.v1_20_R3.command.BukkitCommandWrapper.run(BukkitCommandWrapper.java:64) ~[paper-1.20.4.jar:git-Paper-488]
        at com.mojang.brigadier.context.ContextChain.runExecutable(ContextChain.java:73) ~[brigadier-1.2.9.jar:?]
        at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:31) ~[paper-1.20.4.jar:git-Paper-488]
        at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:19) ~[paper-1.20.4.jar:git-Paper-488]
        at net.minecraft.commands.execution.UnboundEntryAction.lambda$bind$0(UnboundEntryAction.java:8) ~[paper-1.20.4.jar:git-Paper-488]
        at net.minecraft.commands.execution.CommandQueueEntry.a(CommandQueueEntry.java:5) ~[paper-1.20.4.jar:git-Paper-488]
        at net.minecraft.commands.execution.ExecutionContext.runCommandQueue(ExecutionContext.java:103) ~[paper-1.20.4.jar:git-Paper-488]
        at net.minecraft.commands.Commands.executeCommandInContext(Commands.java:434) ~[?:?]
        at net.minecraft.commands.Commands.performCommand(Commands.java:336) ~[?:?]
        at net.minecraft.commands.Commands.performCommand(Commands.java:323) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.performChatCommand(ServerGamePacketListenerImpl.java:2230) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.lambda$handleChatCommand$14(ServerGamePacketListenerImpl.java:2190) ~[?:?]
        at net.minecraft.util.thread.BlockableEventLoop.lambda$submitAsync$0(BlockableEventLoop.java:59) ~[?:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.20.4.jar:git-Paper-488]
        at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:149) ~[?:?]
        at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1465) ~[paper-1.20.4.jar:git-Paper-488]
        at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:194) ~[paper-1.20.4.jar:git-Paper-488]
        at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:123) ~[?:?]
        at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1442) ~[paper-1.20.4.jar:git-Paper-488]
        at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1365) ~[paper-1.20.4.jar:git-Paper-488]
        at net.minecraft.util.thread.BlockableEventLoop.runAllTasks(BlockableEventLoop.java:112) ~[?:?]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1557) ~[paper-1.20.4.jar:git-Paper-488]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1226) ~[paper-1.20.4.jar:git-Paper-488]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[paper-1.20.4.jar:git-Paper-488]
        at java.lang.Thread.run(Thread.java:1570) ~[?:?]
Caused by: java.lang.NoClassDefFoundError: com/craftaro/ultimatestacker/api/utils/Stackable
        at world.bentobox.level.calculators.Pipeliner.addIsland(Pipeliner.java:92) ~[Level-2.13.0-SNAPSHOT-b563.jar:?]
        at world.bentobox.level.LevelsManager.calculateLevel(LevelsManager.java:133) ~[Level-2.13.0-SNAPSHOT-b563.jar:?]
        at world.bentobox.level.commands.IslandLevelCommand.scanIsland(IslandLevelCommand.java:90) ~[Level-2.13.0-SNAPSHOT-b563.jar:?]
        at world.bentobox.level.commands.IslandLevelCommand.execute(IslandLevelCommand.java:70) ~[Level-2.13.0-SNAPSHOT-b563.jar:?]
        at world.bentobox.bentobox.api.commands.CompositeCommand.call(CompositeCommand.java:292) ~[BentoBox-2.3.0.jar:?]
        at world.bentobox.bentobox.api.commands.CompositeCommand.execute(CompositeCommand.java:260) ~[BentoBox-2.3.0.jar:?]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155) ~[paper-api-1.20.4-R0.1-SNAPSHOT.jar:?]
        ... 28 more
```


### Steps/models to reproduce

Latest Jenkins Level.


### BentoBox version

```
> bbox version
[15:14:00 INFO]: Running PAPER 1.20.4.
[15:14:00 INFO]: (git-Paper-488 (MC: 1.20.4))
[15:14:00 INFO]: BentoBox version: 2.3.0
[15:14:00 INFO]: Database: JSON
[15:14:00 INFO]: Loaded Game Worlds:
[15:14:00 INFO]: acid (acid): Overworld
[15:14:00 INFO]: cave (cave): Overworld
[15:14:00 INFO]: oneblock (oneblock): Overworld
[15:14:00 INFO]: parkour (parkour): Overworld
[15:14:00 INFO]: skyblock (skyblock): Overworld
[15:14:00 INFO]: skygrid (skygrid): Overworld
[15:14:00 INFO]: Loaded Addons:
[15:14:00 INFO]: AcidIsland 1.19.0-SNAPSHOT-b1016 (ENABLED)
[15:14:00 INFO]: AOneBlock 1.17.0-SNAPSHOT-b400 (ENABLED)
[15:14:00 INFO]: Biomes 2.2.0-SNAPSHOT-b258 (ENABLED)
[15:14:00 INFO]: Border 4.2.2-SNAPSHOT-b197 (ENABLED)
[15:14:00 INFO]: BSkyBlock 1.18.0-SNAPSHOT-b761 (ENABLED)
[15:14:00 INFO]: CaveBlock 1.19.0-SNAPSHOT-b444 (ENABLED)
[15:14:00 INFO]: Challenges 1.3.0 (ENABLED)
[15:14:00 INFO]: FarmersDance 1.1.0-SNAPSHOT (ENABLED)
[15:14:00 INFO]: Greenhouses 1.8.0-SNAPSHOT-b370 (ENABLED)
[15:14:00 INFO]: Level 2.13.0-SNAPSHOT-b563 (ENABLED)
[15:14:00 INFO]: Limits 1.21.0-SNAPSHOT-b327 (ENABLED)
[15:14:00 INFO]: MagicCobblestoneGenerator 2.5.2-SNAPSHOT-b272 (ENABLED)
[15:14:00 INFO]: Parkour 1.3.1-SNAPSHOT-b86 (ENABLED)
[15:14:00 INFO]: SkyGrid 2.0.0-SNAPSHOT-b426 (ENABLED)
[15:14:00 INFO]: TopBlock 1.0.2-SNAPSHOT-b25 (ENABLED)
[15:14:00 INFO]: Warps 1.15.0-SNAPSHOT-b362 (ENABLED)
>
```


### Plugin list

_No response_

### Other

_No response_

--- a/src/main/java/world/bentobox/level/calculators/IslandLevelCalculator.java
+++ b/src/main/java/world/bentobox/level/calculators/IslandLevelCalculator.java
@@ -40,8 +40,6 @@ import com.bgsoftware.wildstacker.api.objects.StackedBarrel;
 import com.google.common.collect.Multiset;
 import com.google.common.collect.Multiset.Entry;
 import com.google.common.collect.Multisets;
-import com.craftaro.ultimatestacker.api.UltimateStackerApi;
-import com.craftaro.ultimatestacker.api.utils.Stackable;
 
 import dev.rosewood.rosestacker.api.RoseStackerAPI;
 import us.lynuxcraft.deadsilenceiv.advancedchests.AdvancedChestsAPI;
@@ -58,13 +56,13 @@ public class IslandLevelCalculator {
     private static final String LINE_BREAK = "==================================";
     public static final long MAX_AMOUNT = 10000000;
     private static final List<Material> CHESTS = Arrays.asList(Material.CHEST, Material.CHEST_MINECART,
-	    Material.TRAPPED_CHEST, Material.SHULKER_BOX, Material.BLACK_SHULKER_BOX, Material.BLUE_SHULKER_BOX,
-	    Material.BROWN_SHULKER_BOX, Material.CYAN_SHULKER_BOX, Material.GRAY_SHULKER_BOX,
-	    Material.GREEN_SHULKER_BOX, Material.LIGHT_BLUE_SHULKER_BOX, Material.LIGHT_GRAY_SHULKER_BOX,
-	    Material.LIME_SHULKER_BOX, Material.MAGENTA_SHULKER_BOX, Material.ORANGE_SHULKER_BOX,
-	    Material.PINK_SHULKER_BOX, Material.PURPLE_SHULKER_BOX, Material.RED_SHULKER_BOX, Material.RED_SHULKER_BOX,
-	    Material.WHITE_SHULKER_BOX, Material.YELLOW_SHULKER_BOX, Material.COMPOSTER, Material.BARREL,
-	    Material.DISPENSER, Material.DROPPER, Material.SMOKER, Material.BLAST_FURNACE);
+            Material.TRAPPED_CHEST, Material.SHULKER_BOX, Material.BLACK_SHULKER_BOX, Material.BLUE_SHULKER_BOX,
+            Material.BROWN_SHULKER_BOX, Material.CYAN_SHULKER_BOX, Material.GRAY_SHULKER_BOX,
+            Material.GREEN_SHULKER_BOX, Material.LIGHT_BLUE_SHULKER_BOX, Material.LIGHT_GRAY_SHULKER_BOX,
+            Material.LIME_SHULKER_BOX, Material.MAGENTA_SHULKER_BOX, Material.ORANGE_SHULKER_BOX,
+            Material.PINK_SHULKER_BOX, Material.PURPLE_SHULKER_BOX, Material.RED_SHULKER_BOX, Material.RED_SHULKER_BOX,
+            Material.WHITE_SHULKER_BOX, Material.YELLOW_SHULKER_BOX, Material.COMPOSTER, Material.BARREL,
+            Material.DISPENSER, Material.DROPPER, Material.SMOKER, Material.BLAST_FURNACE);
     private static final int CHUNKS_TO_SCAN = 100;
     private final Level addon;
     private final Queue<Pair<Integer, Integer>> chunksToCheck;
@@ -91,34 +89,34 @@ public class IslandLevelCalculator {
      * @param zeroIsland - true if the calculation is due to an island zeroing
      */
     public IslandLevelCalculator(Level addon, Island island, CompletableFuture<Results> r, boolean zeroIsland) {
-	this.addon = addon;
-	this.island = island;
-	this.r = r;
-	this.zeroIsland = zeroIsland;
-	results = new Results();
-	duration = System.currentTimeMillis();
-	chunksToCheck = getChunksToScan(island);
-	this.limitCount = new EnumMap<>(addon.getBlockConfig().getBlockLimits());
-	// Get the initial island level
-	results.initialLevel.set(addon.getInitialIslandLevel(island));
-	// Set up the worlds
-	worlds.put(Environment.NORMAL, Util.getWorld(island.getWorld()));
-	// Nether
-	if (addon.getSettings().isNether()) {
-	    World nether = addon.getPlugin().getIWM().getNetherWorld(island.getWorld());
-	    if (nether != null) {
-		worlds.put(Environment.NETHER, nether);
-	    }
-	}
-	// End
-	if (addon.getSettings().isEnd()) {
-	    World end = addon.getPlugin().getIWM().getEndWorld(island.getWorld());
-	    if (end != null) {
-		worlds.put(Environment.THE_END, end);
-	    }
-	}
-	// Sea Height
-	seaHeight = addon.getPlugin().getIWM().getSeaHeight(island.getWorld());
+        this.addon = addon;
+        this.island = island;
+        this.r = r;
+        this.zeroIsland = zeroIsland;
+        results = new Results();
+        duration = System.currentTimeMillis();
+        chunksToCheck = getChunksToScan(island);
+        this.limitCount = new EnumMap<>(addon.getBlockConfig().getBlockLimits());
+        // Get the initial island level
+        results.initialLevel.set(addon.getInitialIslandLevel(island));
+        // Set up the worlds
+        worlds.put(Environment.NORMAL, Util.getWorld(island.getWorld()));
+        // Nether
+        if (addon.getSettings().isNether()) {
+            World nether = addon.getPlugin().getIWM().getNetherWorld(island.getWorld());
+            if (nether != null) {
+                worlds.put(Environment.NETHER, nether);
+            }
+        }
+        // End
+        if (addon.getSettings().isEnd()) {
+            World end = addon.getPlugin().getIWM().getEndWorld(island.getWorld());
+            if (end != null) {
+                worlds.put(Environment.THE_END, end);
+            }
+        }
+        // Sea Height
+        seaHeight = addon.getPlugin().getIWM().getSeaHeight(island.getWorld());
     }
 
     /**
@@ -128,18 +126,18 @@ public class IslandLevelCalculator {
      * @return level of island
      */
     private long calculateLevel(long blockAndDeathPoints) {
-	String calcString = addon.getSettings().getLevelCalc();
-	String withValues = calcString.replace("blocks", String.valueOf(blockAndDeathPoints)).replace("level_cost",
-		String.valueOf(this.addon.getSettings().getLevelCost()));
-	long evalWithValues;
-	try {
-	    evalWithValues = (long) EquationEvaluator.eval(withValues);
-	    return evalWithValues - (addon.getSettings().isZeroNewIslandLevels() ? results.initialLevel.get() : 0);
+        String calcString = addon.getSettings().getLevelCalc();
+        String withValues = calcString.replace("blocks", String.valueOf(blockAndDeathPoints)).replace("level_cost",
+                String.valueOf(this.addon.getSettings().getLevelCost()));
+        long evalWithValues;
+        try {
+            evalWithValues = (long) EquationEvaluator.eval(withValues);
+            return evalWithValues - (addon.getSettings().isZeroNewIslandLevels() ? results.initialLevel.get() : 0);
 
-	} catch (ParseException e) {
-	    addon.getPlugin().logStacktrace(e);
-	    return 0L;
-	}
+        } catch (ParseException e) {
+            addon.getPlugin().logStacktrace(e);
+            return 0L;
+        }
     }
 
     /**
@@ -150,14 +148,14 @@ public class IslandLevelCalculator {
      * @param belowSeaLevel - true if below sea level
      */
     private void checkBlock(Material mat, boolean belowSeaLevel) {
-	int count = limitCount(mat);
-	if (belowSeaLevel) {
-	    results.underWaterBlockCount.addAndGet(count);
-	    results.uwCount.add(mat);
-	} else {
-	    results.rawBlockCount.addAndGet(count);
-	    results.mdCount.add(mat);
-	}
+        int count = limitCount(mat);
+        if (belowSeaLevel) {
+            results.underWaterBlockCount.addAndGet(count);
+            results.uwCount.add(mat);
+        } else {
+            results.rawBlockCount.addAndGet(count);
+            results.mdCount.add(mat);
+        }
     }
 
     /**
@@ -167,22 +165,22 @@ public class IslandLevelCalculator {
      * @return - set of pairs of x,z coordinates to check
      */
     private Queue<Pair<Integer, Integer>> getChunksToScan(Island island) {
-	Queue<Pair<Integer, Integer>> chunkQueue = new ConcurrentLinkedQueue<>();
-	for (int x = island.getMinProtectedX(); x < (island.getMinProtectedX() + island.getProtectionRange() * 2
-		+ 16); x += 16) {
-	    for (int z = island.getMinProtectedZ(); z < (island.getMinProtectedZ() + island.getProtectionRange() * 2
-		    + 16); z += 16) {
-		chunkQueue.add(new Pair<>(x >> 4, z >> 4));
-	    }
-	}
-	return chunkQueue;
+        Queue<Pair<Integer, Integer>> chunkQueue = new ConcurrentLinkedQueue<>();
+        for (int x = island.getMinProtectedX(); x < (island.getMinProtectedX() + island.getProtectionRange() * 2
+                + 16); x += 16) {
+            for (int z = island.getMinProtectedZ(); z < (island.getMinProtectedZ() + island.getProtectionRange() * 2
+                    + 16); z += 16) {
+                chunkQueue.add(new Pair<>(x >> 4, z >> 4));
+            }
+        }
+        return chunkQueue;
     }
 
     /**
      * @return the island
      */
     public Island getIsland() {
-	return island;
+        return island;
     }
 
     /**
@@ -191,7 +189,7 @@ public class IslandLevelCalculator {
      * @return the r
      */
     public CompletableFuture<Results> getR() {
-	return r;
+        return r;
     }
 
     /**
@@ -200,67 +198,67 @@ public class IslandLevelCalculator {
      * @return a list of lines
      */
     private List<String> getReport() {
-	List<String> reportLines = new ArrayList<>();
-	// provide counts
-	reportLines.add("Level Log for island in " + addon.getPlugin().getIWM().getFriendlyName(island.getWorld())
-		+ " at " + Util.xyz(island.getCenter().toVector()));
-	reportLines.add("Island owner UUID = " + island.getOwner());
-	reportLines.add("Total block value count = " + String.format("%,d", results.rawBlockCount.get()));
-	reportLines.add("Formula to calculate island level: " + addon.getSettings().getLevelCalc());
-	reportLines.add("Level cost = " + addon.getSettings().getLevelCost());
-	reportLines.add("Deaths handicap = " + results.deathHandicap.get());
-	if (addon.getSettings().isZeroNewIslandLevels()) {
-	    reportLines.add("Initial island level = " + (0L - addon.getManager().getInitialLevel(island)));
-	}
-	reportLines.add("Previous level = " + addon.getManager().getIslandLevel(island.getWorld(), island.getOwner()));
-	reportLines.add("New level = " + results.getLevel());
-	reportLines.add(LINE_BREAK);
-	int total = 0;
-	if (!results.uwCount.isEmpty()) {
-	    reportLines.add("Underwater block count (Multiplier = x" + addon.getSettings().getUnderWaterMultiplier()
-		    + ") value");
-	    reportLines.add("Total number of underwater blocks = " + String.format("%,d", results.uwCount.size()));
-	    reportLines.addAll(sortedReport(total, results.uwCount));
-	}
-	reportLines.add("Regular block count");
-	reportLines.add("Total number of blocks = " + String.format("%,d", results.mdCount.size()));
-	reportLines.addAll(sortedReport(total, results.mdCount));
+        List<String> reportLines = new ArrayList<>();
+        // provide counts
+        reportLines.add("Level Log for island in " + addon.getPlugin().getIWM().getFriendlyName(island.getWorld())
+                + " at " + Util.xyz(island.getCenter().toVector()));
+        reportLines.add("Island owner UUID = " + island.getOwner());
+        reportLines.add("Total block value count = " + String.format("%,d", results.rawBlockCount.get()));
+        reportLines.add("Formula to calculate island level: " + addon.getSettings().getLevelCalc());
+        reportLines.add("Level cost = " + addon.getSettings().getLevelCost());
+        reportLines.add("Deaths handicap = " + results.deathHandicap.get());
+        if (addon.getSettings().isZeroNewIslandLevels()) {
+            reportLines.add("Initial island level = " + (0L - addon.getManager().getInitialLevel(island)));
+        }
+        reportLines.add("Previous level = " + addon.getManager().getIslandLevel(island.getWorld(), island.getOwner()));
+        reportLines.add("New level = " + results.getLevel());
+        reportLines.add(LINE_BREAK);
+        int total = 0;
+        if (!results.uwCount.isEmpty()) {
+            reportLines.add("Underwater block count (Multiplier = x" + addon.getSettings().getUnderWaterMultiplier()
+                    + ") value");
+            reportLines.add("Total number of underwater blocks = " + String.format("%,d", results.uwCount.size()));
+            reportLines.addAll(sortedReport(total, results.uwCount));
+        }
+        reportLines.add("Regular block count");
+        reportLines.add("Total number of blocks = " + String.format("%,d", results.mdCount.size()));
+        reportLines.addAll(sortedReport(total, results.mdCount));
 
-	reportLines.add(
-		"Blocks not counted because they exceeded limits: " + String.format("%,d", results.ofCount.size()));
-	Iterable<Multiset.Entry<Material>> entriesSortedByCount = results.ofCount.entrySet();
-	Iterator<Entry<Material>> it = entriesSortedByCount.iterator();
-	while (it.hasNext()) {
-	    Entry<Material> type = it.next();
-	    Integer limit = addon.getBlockConfig().getBlockLimits().get(type.getElement());
-	    String explain = ")";
-	    if (limit == null) {
-		Material generic = type.getElement();
-		limit = addon.getBlockConfig().getBlockLimits().get(generic);
-		explain = " - All types)";
-	    }
-	    reportLines.add(type.getElement().toString() + ": " + String.format("%,d", type.getCount())
-		    + " blocks (max " + limit + explain);
-	}
-	reportLines.add(LINE_BREAK);
-	reportLines.add("Blocks on island that are not in config.yml");
-	reportLines.add("Total number = " + String.format("%,d", results.ncCount.size()));
-	entriesSortedByCount = results.ncCount.entrySet();
-	it = entriesSortedByCount.iterator();
-	while (it.hasNext()) {
-	    Entry<Material> type = it.next();
-	    reportLines.add(type.getElement().toString() + ": " + String.format("%,d", type.getCount()) + " blocks");
-	}
-	reportLines.add(LINE_BREAK);
+        reportLines.add(
+                "Blocks not counted because they exceeded limits: " + String.format("%,d", results.ofCount.size()));
+        Iterable<Multiset.Entry<Material>> entriesSortedByCount = results.ofCount.entrySet();
+        Iterator<Entry<Material>> it = entriesSortedByCount.iterator();
+        while (it.hasNext()) {
+            Entry<Material> type = it.next();
+            Integer limit = addon.getBlockConfig().getBlockLimits().get(type.getElement());
+            String explain = ")";
+            if (limit == null) {
+                Material generic = type.getElement();
+                limit = addon.getBlockConfig().getBlockLimits().get(generic);
+                explain = " - All types)";
+            }
+            reportLines.add(type.getElement().toString() + ": " + String.format("%,d", type.getCount())
+                    + " blocks (max " + limit + explain);
+        }
+        reportLines.add(LINE_BREAK);
+        reportLines.add("Blocks on island that are not in config.yml");
+        reportLines.add("Total number = " + String.format("%,d", results.ncCount.size()));
+        entriesSortedByCount = results.ncCount.entrySet();
+        it = entriesSortedByCount.iterator();
+        while (it.hasNext()) {
+            Entry<Material> type = it.next();
+            reportLines.add(type.getElement().toString() + ": " + String.format("%,d", type.getCount()) + " blocks");
+        }
+        reportLines.add(LINE_BREAK);
 
-	return reportLines;
+        return reportLines;
     }
 
     /**
      * @return the results
      */
     public Results getResults() {
-	return results;
+        return results;
     }
 
     /**
@@ -270,13 +268,13 @@ public class IslandLevelCalculator {
      * @return value of a material
      */
     private int getValue(Material md) {
-	Integer value = addon.getBlockConfig().getValue(island.getWorld(), md);
-	if (value == null) {
-	    // Not in config
-	    results.ncCount.add(md);
-	    return 0;
-	}
-	return value;
+        Integer value = addon.getBlockConfig().getValue(island.getWorld(), md);
+        if (value == null) {
+            // Not in config
+            results.ncCount.add(md);
+            return 0;
+        }
+        return value;
     }
 
     /**
@@ -288,44 +286,44 @@ public class IslandLevelCalculator {
      *         there is no island nether
      */
     private CompletableFuture<List<Chunk>> getWorldChunk(Environment env, Queue<Pair<Integer, Integer>> pairList) {
-	if (worlds.containsKey(env)) {
-	    CompletableFuture<List<Chunk>> r2 = new CompletableFuture<>();
-	    List<Chunk> chunkList = new ArrayList<>();
-	    World world = worlds.get(env);
-	    // Get the chunk, and then coincidentally check the RoseStacker
-	    loadChunks(r2, world, pairList, chunkList);
-	    return r2;
-	}
-	return CompletableFuture.completedFuture(Collections.emptyList());
+        if (worlds.containsKey(env)) {
+            CompletableFuture<List<Chunk>> r2 = new CompletableFuture<>();
+            List<Chunk> chunkList = new ArrayList<>();
+            World world = worlds.get(env);
+            // Get the chunk, and then coincidentally check the RoseStacker
+            loadChunks(r2, world, pairList, chunkList);
+            return r2;
+        }
+        return CompletableFuture.completedFuture(Collections.emptyList());
     }
 
     private void loadChunks(CompletableFuture<List<Chunk>> r2, World world, Queue<Pair<Integer, Integer>> pairList,
-	    List<Chunk> chunkList) {
-	if (pairList.isEmpty()) {
-	    r2.complete(chunkList);
-	    return;
-	}
-	Pair<Integer, Integer> p = pairList.poll();
-	Util.getChunkAtAsync(world, p.x, p.z, world.getEnvironment().equals(Environment.NETHER)).thenAccept(chunk -> {
-	    if (chunk != null) {
-		chunkList.add(chunk);
-		roseStackerCheck(chunk);
-	    }
-	    loadChunks(r2, world, pairList, chunkList); // Iteration
-	});
+            List<Chunk> chunkList) {
+        if (pairList.isEmpty()) {
+            r2.complete(chunkList);
+            return;
+        }
+        Pair<Integer, Integer> p = pairList.poll();
+        Util.getChunkAtAsync(world, p.x, p.z, world.getEnvironment().equals(Environment.NETHER)).thenAccept(chunk -> {
+            if (chunk != null) {
+                chunkList.add(chunk);
+                roseStackerCheck(chunk);
+            }
+            loadChunks(r2, world, pairList, chunkList); // Iteration
+        });
     }
 
     private void roseStackerCheck(Chunk chunk) {
-	if (addon.isRoseStackersEnabled()) {
-	    RoseStackerAPI.getInstance().getStackedBlocks(Collections.singletonList(chunk)).forEach(e -> {
-		// Blocks below sea level can be scored differently
-		boolean belowSeaLevel = seaHeight > 0 && e.getLocation().getY() <= seaHeight;
-		// Check block once because the base block will be counted in the chunk snapshot
-		for (int _x = 0; _x < e.getStackSize() - 1; _x++) {
-		    checkBlock(e.getBlock().getType(), belowSeaLevel);
-		}
-	    });
-	}
+        if (addon.isRoseStackersEnabled()) {
+            RoseStackerAPI.getInstance().getStackedBlocks(Collections.singletonList(chunk)).forEach(e -> {
+                // Blocks below sea level can be scored differently
+                boolean belowSeaLevel = seaHeight > 0 && e.getLocation().getY() <= seaHeight;
+                // Check block once because the base block will be counted in the chunk snapshot
+                for (int _x = 0; _x < e.getStackSize() - 1; _x++) {
+                    checkBlock(e.getBlock().getType(), belowSeaLevel);
+                }
+            });
+        }
     }
 
     /**
@@ -336,17 +334,17 @@ public class IslandLevelCalculator {
      * @return value of the block if can be counted
      */
     private int limitCount(Material md) {
-	if (limitCount.containsKey(md)) {
-	    int count = limitCount.get(md);
-	    if (count > 0) {
-		limitCount.put(md, --count);
-		return getValue(md);
-	    } else {
-		results.ofCount.add(md);
-		return 0;
-	    }
-	}
-	return getValue(md);
+        if (limitCount.containsKey(md)) {
+            int count = limitCount.get(md);
+            if (count > 0) {
+                limitCount.put(md, --count);
+                return getValue(md);
+            } else {
+                results.ofCount.add(md);
+                return 0;
+            }
+        }
+        return getValue(md);
     }
 
     /**
@@ -355,39 +353,39 @@ public class IslandLevelCalculator {
      * @param chunk - the chunk to scan
      */
     private void scanChests(Chunk chunk) {
-	// Count blocks in chests
-	for (BlockState bs : chunk.getTileEntities()) {
-	    if (bs instanceof Container container) {
-		if (addon.isAdvChestEnabled()) {
-		    AdvancedChest<?, ?> aChest = AdvancedChestsAPI.getChestManager().getAdvancedChest(bs.getLocation());
-		    if (aChest != null && aChest.getChestType().getName().equals("NORMAL")) {
-			aChest.getPages().stream().map(ChestPage::getItems).forEach(c -> {
-			    for (Object i : c) {
-				countItemStack((ItemStack) i);
-			    }
-			});
-			continue;
-		    }
-		}
-		// Regular chest
-		container.getSnapshotInventory().forEach(this::countItemStack);
-	    }
-	}
+        // Count blocks in chests
+        for (BlockState bs : chunk.getTileEntities()) {
+            if (bs instanceof Container container) {
+                if (addon.isAdvChestEnabled()) {
+                    AdvancedChest<?, ?> aChest = AdvancedChestsAPI.getChestManager().getAdvancedChest(bs.getLocation());
+                    if (aChest != null && aChest.getChestType().getName().equals("NORMAL")) {
+                        aChest.getPages().stream().map(ChestPage::getItems).forEach(c -> {
+                            for (Object i : c) {
+                                countItemStack((ItemStack) i);
+                            }
+                        });
+                        continue;
+                    }
+                }
+                // Regular chest
+                container.getSnapshotInventory().forEach(this::countItemStack);
+            }
+        }
     }
 
     private void countItemStack(ItemStack i) {
-	if (i == null || !i.getType().isBlock())
-	    return;
+        if (i == null || !i.getType().isBlock())
+            return;
 
-	for (int c = 0; c < i.getAmount(); c++) {
-	    if (addon.getSettings().isIncludeShulkersInChest()
-		    && i.getItemMeta() instanceof BlockStateMeta blockStateMeta
-		    && blockStateMeta.getBlockState() instanceof ShulkerBox shulkerBox) {
-		shulkerBox.getSnapshotInventory().forEach(this::countItemStack);
-	    }
+        for (int c = 0; c < i.getAmount(); c++) {
+            if (addon.getSettings().isIncludeShulkersInChest()
+                    && i.getItemMeta() instanceof BlockStateMeta blockStateMeta
+                    && blockStateMeta.getBlockState() instanceof ShulkerBox shulkerBox) {
+                shulkerBox.getSnapshotInventory().forEach(this::countItemStack);
+            }
 
-	    checkBlock(i.getType(), false);
-	}
+            checkBlock(i.getType(), false);
+        }
     }
 
     /**
@@ -400,26 +398,26 @@ public class IslandLevelCalculator {
      *         that will be true if the scan was successful, false if not
      */
     private CompletableFuture<Boolean> scanChunk(List<Chunk> chunks) {
-	// If the chunk hasn't been generated, return
-	if (chunks == null || chunks.isEmpty()) {
-	    return CompletableFuture.completedFuture(false);
-	}
-	// Count blocks in chunk
-	CompletableFuture<Boolean> result = new CompletableFuture<>();
-	/*
-	 * At this point, we need to grab a snapshot of each chunk and then scan it
-	 * async. At the end, we make the CompletableFuture true to show it is done. I'm
-	 * not sure how much lag this will cause, but as all the chunks are loaded,
-	 * maybe not that much.
-	 */
-	List<ChunkPair> preLoad = chunks.stream().map(c -> new ChunkPair(c.getWorld(), c, c.getChunkSnapshot()))
-		.toList();
-	Bukkit.getScheduler().runTaskAsynchronously(BentoBox.getInstance(), () -> {
-	    preLoad.forEach(this::scanAsync);
-	    // Once they are all done, return to the main thread.
-	    Bukkit.getScheduler().runTask(addon.getPlugin(), () -> result.complete(true));
-	});
-	return result;
+        // If the chunk hasn't been generated, return
+        if (chunks == null || chunks.isEmpty()) {
+            return CompletableFuture.completedFuture(false);
+        }
+        // Count blocks in chunk
+        CompletableFuture<Boolean> result = new CompletableFuture<>();
+        /*
+         * At this point, we need to grab a snapshot of each chunk and then scan it
+         * async. At the end, we make the CompletableFuture true to show it is done. I'm
+         * not sure how much lag this will cause, but as all the chunks are loaded,
+         * maybe not that much.
+         */
+        List<ChunkPair> preLoad = chunks.stream().map(c -> new ChunkPair(c.getWorld(), c, c.getChunkSnapshot()))
+                .toList();
+        Bukkit.getScheduler().runTaskAsynchronously(BentoBox.getInstance(), () -> {
+            preLoad.forEach(this::scanAsync);
+            // Once they are all done, return to the main thread.
+            Bukkit.getScheduler().runTask(addon.getPlugin(), () -> result.complete(true));
+        });
+        return result;
     }
 
     record ChunkPair(World world, Chunk chunk, ChunkSnapshot chunkSnapshot) {
@@ -431,66 +429,53 @@ public class IslandLevelCalculator {
      * @param cp chunk to scan
      */
     private void scanAsync(ChunkPair cp) {
-	for (int x = 0; x < 16; x++) {
-	    // Check if the block coordinate is inside the protection zone and if not, don't
-	    // count it
-	    if (cp.chunkSnapshot.getX() * 16 + x < island.getMinProtectedX() || cp.chunkSnapshot.getX() * 16
-		    + x >= island.getMinProtectedX() + island.getProtectionRange() * 2) {
-		continue;
-	    }
-	    for (int z = 0; z < 16; z++) {
-		// Check if the block coordinate is inside the protection zone and if not, don't
-		// count it
-		if (cp.chunkSnapshot.getZ() * 16 + z < island.getMinProtectedZ() || cp.chunkSnapshot.getZ() * 16
-			+ z >= island.getMinProtectedZ() + island.getProtectionRange() * 2) {
-		    continue;
-		}
-		// Only count to the highest block in the world for some optimization
-		for (int y = cp.world.getMinHeight(); y < cp.world.getMaxHeight(); y++) {
-		    BlockData blockData = cp.chunkSnapshot.getBlockData(x, y, z);
-		    boolean belowSeaLevel = seaHeight > 0 && y <= seaHeight;
-		    // Slabs can be doubled, so check them twice
-		    if (Tag.SLABS.isTagged(blockData.getMaterial())) {
-			Slab slab = (Slab) blockData;
-			if (slab.getType().equals(Slab.Type.DOUBLE)) {
-			    checkBlock(blockData.getMaterial(), belowSeaLevel);
-			}
-		    }
-		    // Hook for Wild Stackers (Blocks and Spawners Only) - this has to use the real
-		    // chunk
-		    if (addon.isStackersEnabled() && (blockData.getMaterial().equals(Material.CAULDRON)
-			    || blockData.getMaterial().equals(Material.SPAWNER))) {
-			stackedBlocks.add(new Location(cp.world, (double) x + cp.chunkSnapshot.getX() * 16, y,
-				(double) z + cp.chunkSnapshot.getZ() * 16));
-		    }
+        for (int x = 0; x < 16; x++) {
+            // Check if the block coordinate is inside the protection zone and if not, don't
+            // count it
+            if (cp.chunkSnapshot.getX() * 16 + x < island.getMinProtectedX() || cp.chunkSnapshot.getX() * 16
+                    + x >= island.getMinProtectedX() + island.getProtectionRange() * 2) {
+                continue;
+            }
+            for (int z = 0; z < 16; z++) {
+                // Check if the block coordinate is inside the protection zone and if not, don't
+                // count it
+                if (cp.chunkSnapshot.getZ() * 16 + z < island.getMinProtectedZ() || cp.chunkSnapshot.getZ() * 16
+                        + z >= island.getMinProtectedZ() + island.getProtectionRange() * 2) {
+                    continue;
+                }
+                // Only count to the highest block in the world for some optimization
+                for (int y = cp.world.getMinHeight(); y < cp.world.getMaxHeight(); y++) {
+                    BlockData blockData = cp.chunkSnapshot.getBlockData(x, y, z);
+                    Material m = blockData.getMaterial();
+                    boolean belowSeaLevel = seaHeight > 0 && y <= seaHeight;
+                    // Slabs can be doubled, so check them twice
+                    if (Tag.SLABS.isTagged(m)) {
+                        Slab slab = (Slab) blockData;
+                        if (slab.getType().equals(Slab.Type.DOUBLE)) {
+                            checkBlock(m, belowSeaLevel);
+                        }
+                    }
+                    // Hook for Wild Stackers (Blocks and Spawners Only) - this has to use the real
+                    // chunk
+                    if (addon.isStackersEnabled() && (m.equals(Material.CAULDRON) || m.equals(Material.SPAWNER))) {
+                        stackedBlocks.add(new Location(cp.world, (double) x + cp.chunkSnapshot.getX() * 16, y,
+                                (double) z + cp.chunkSnapshot.getZ() * 16));
+                    }
 
-		    Block block = cp.chunk.getBlock(x, y, z);
+                    if (addon.isUltimateStackerEnabled() && !m.isAir()) {
+                        Location l = new Location(cp.chunk.getWorld(), x, y, z);
+                        UltimateStackerCalc.addStackers(m, l, results, belowSeaLevel, limitCount(m));
+                    }
 
-		    if (addon.isUltimateStackerEnabled()) {
-			if (!blockData.getMaterial().equals(Material.AIR)) {
-			    Stackable stack = UltimateStackerApi.getBlockStackManager().getBlock(block.getLocation());
-			    if (stack != null) {
-				int value = limitCount(blockData.getMaterial());
-				if (belowSeaLevel) {
-				    results.underWaterBlockCount.addAndGet((long) stack.getAmount() * value);
-				    results.uwCount.add(blockData.getMaterial());
-				} else {
-				    results.rawBlockCount.addAndGet((long) stack.getAmount() * value);
-				    results.mdCount.add(blockData.getMaterial());
-				}
-			    }
-			}
-		    }
-
-		    // Scan chests
-		    if (addon.getSettings().isIncludeChests() && CHESTS.contains(blockData.getMaterial())) {
-			chestBlocks.add(cp.chunk);
-		    }
-		    // Add the value of the block's material
-		    checkBlock(blockData.getMaterial(), belowSeaLevel);
-		}
-	    }
-	}
+                    // Scan chests
+                    if (addon.getSettings().isIncludeChests() && CHESTS.contains(m)) {
+                        chestBlocks.add(cp.chunk);
+                    }
+                    // Add the value of the block's material
+                    checkBlock(m, belowSeaLevel);
+                }
+            }
+        }
     }
 
     /**
@@ -500,180 +485,180 @@ public class IslandLevelCalculator {
      *         to be scanned, and false if not
      */
     public CompletableFuture<Boolean> scanNextChunk() {
-	if (chunksToCheck.isEmpty()) {
-	    addon.logError("Unexpected: no chunks to scan!");
-	    // This should not be needed, but just in case
-	    return CompletableFuture.completedFuture(false);
-	}
-	// Retrieve and remove from the queue
-	Queue<Pair<Integer, Integer>> pairList = new ConcurrentLinkedQueue<>();
-	int i = 0;
-	while (!chunksToCheck.isEmpty() && i++ < CHUNKS_TO_SCAN) {
-	    pairList.add(chunksToCheck.poll());
-	}
-	Queue<Pair<Integer, Integer>> endPairList = new ConcurrentLinkedQueue<>(pairList);
-	Queue<Pair<Integer, Integer>> netherPairList = new ConcurrentLinkedQueue<>(pairList);
-	// Set up the result
-	CompletableFuture<Boolean> result = new CompletableFuture<>();
-	// Get chunks and scan
-	// Get chunks and scan
-	getWorldChunk(Environment.THE_END, endPairList).thenAccept(
-		endChunks -> scanChunk(endChunks).thenAccept(b -> getWorldChunk(Environment.NETHER, netherPairList)
-			.thenAccept(netherChunks -> scanChunk(netherChunks)
-				.thenAccept(b2 -> getWorldChunk(Environment.NORMAL, pairList)
-					.thenAccept(normalChunks -> scanChunk(normalChunks).thenAccept(b3 ->
-					// Complete the result now that all chunks have been scanned
-					result.complete(!chunksToCheck.isEmpty())))))));
+        if (chunksToCheck.isEmpty()) {
+            addon.logError("Unexpected: no chunks to scan!");
+            // This should not be needed, but just in case
+            return CompletableFuture.completedFuture(false);
+        }
+        // Retrieve and remove from the queue
+        Queue<Pair<Integer, Integer>> pairList = new ConcurrentLinkedQueue<>();
+        int i = 0;
+        while (!chunksToCheck.isEmpty() && i++ < CHUNKS_TO_SCAN) {
+            pairList.add(chunksToCheck.poll());
+        }
+        Queue<Pair<Integer, Integer>> endPairList = new ConcurrentLinkedQueue<>(pairList);
+        Queue<Pair<Integer, Integer>> netherPairList = new ConcurrentLinkedQueue<>(pairList);
+        // Set up the result
+        CompletableFuture<Boolean> result = new CompletableFuture<>();
+        // Get chunks and scan
+        // Get chunks and scan
+        getWorldChunk(Environment.THE_END, endPairList).thenAccept(
+                endChunks -> scanChunk(endChunks).thenAccept(b -> getWorldChunk(Environment.NETHER, netherPairList)
+                        .thenAccept(netherChunks -> scanChunk(netherChunks)
+                                .thenAccept(b2 -> getWorldChunk(Environment.NORMAL, pairList)
+                                        .thenAccept(normalChunks -> scanChunk(normalChunks).thenAccept(b3 ->
+                                        // Complete the result now that all chunks have been scanned
+                                        result.complete(!chunksToCheck.isEmpty())))))));
 
-	return result;
+        return result;
     }
 
     private Collection<String> sortedReport(int total, Multiset<Material> materialCount) {
-	Collection<String> result = new ArrayList<>();
-	Iterable<Multiset.Entry<Material>> entriesSortedByCount = Multisets.copyHighestCountFirst(materialCount)
-		.entrySet();
-	for (Entry<Material> en : entriesSortedByCount) {
-	    Material type = en.getElement();
+        Collection<String> result = new ArrayList<>();
+        Iterable<Multiset.Entry<Material>> entriesSortedByCount = Multisets.copyHighestCountFirst(materialCount)
+                .entrySet();
+        for (Entry<Material> en : entriesSortedByCount) {
+            Material type = en.getElement();
 
-	    int value = getValue(type);
+            int value = getValue(type);
 
-	    result.add(type.toString() + ":" + String.format("%,d", en.getCount()) + " blocks x " + value + " = "
-		    + (value * en.getCount()));
-	    total += (value * en.getCount());
+            result.add(type.toString() + ":" + String.format("%,d", en.getCount()) + " blocks x " + value + " = "
+                    + (value * en.getCount()));
+            total += (value * en.getCount());
 
-	}
-	result.add("Subtotal = " + total);
-	result.add(LINE_BREAK);
-	return result;
+        }
+        result.add("Subtotal = " + total);
+        result.add(LINE_BREAK);
+        return result;
     }
 
     /**
      * Finalizes the calculations and makes the report
      */
     public void tidyUp() {
-	// Finalize calculations
-	results.rawBlockCount
-		.addAndGet((long) (results.underWaterBlockCount.get() * addon.getSettings().getUnderWaterMultiplier()));
+        // Finalize calculations
+        results.rawBlockCount
+                .addAndGet((long) (results.underWaterBlockCount.get() * addon.getSettings().getUnderWaterMultiplier()));
 
-	// Set the death penalty
-	if (this.addon.getSettings().isSumTeamDeaths()) {
-	    for (UUID uuid : this.island.getMemberSet()) {
-		this.results.deathHandicap.addAndGet(this.addon.getPlayers().getDeaths(island.getWorld(), uuid));
-	    }
-	} else {
-	    // At this point, it may be that the island has become unowned.
-	    this.results.deathHandicap.set(this.island.getOwner() == null ? 0
-		    : this.addon.getPlayers().getDeaths(island.getWorld(), this.island.getOwner()));
-	}
+        // Set the death penalty
+        if (this.addon.getSettings().isSumTeamDeaths()) {
+            for (UUID uuid : this.island.getMemberSet()) {
+                this.results.deathHandicap.addAndGet(this.addon.getPlayers().getDeaths(island.getWorld(), uuid));
+            }
+        } else {
+            // At this point, it may be that the island has become unowned.
+            this.results.deathHandicap.set(this.island.getOwner() == null ? 0
+                    : this.addon.getPlayers().getDeaths(island.getWorld(), this.island.getOwner()));
+        }
 
-	long blockAndDeathPoints = this.results.rawBlockCount.get();
-	this.results.totalPoints.set(blockAndDeathPoints);
+        long blockAndDeathPoints = this.results.rawBlockCount.get();
+        this.results.totalPoints.set(blockAndDeathPoints);
 
-	if (this.addon.getSettings().getDeathPenalty() > 0) {
-	    // Proper death penalty calculation.
-	    blockAndDeathPoints -= this.results.deathHandicap.get() * this.addon.getSettings().getDeathPenalty();
-	}
-	this.results.level.set(calculateLevel(blockAndDeathPoints));
+        if (this.addon.getSettings().getDeathPenalty() > 0) {
+            // Proper death penalty calculation.
+            blockAndDeathPoints -= this.results.deathHandicap.get() * this.addon.getSettings().getDeathPenalty();
+        }
+        this.results.level.set(calculateLevel(blockAndDeathPoints));
 
-	// Calculate how many points are required to get to the next level
-	long nextLevel = this.results.level.get();
-	long blocks = blockAndDeathPoints;
-	while (nextLevel < this.results.level.get() + 1 && blocks - blockAndDeathPoints < MAX_AMOUNT) {
-	    nextLevel = calculateLevel(++blocks);
-	}
-	this.results.pointsToNextLevel.set(blocks - blockAndDeathPoints);
+        // Calculate how many points are required to get to the next level
+        long nextLevel = this.results.level.get();
+        long blocks = blockAndDeathPoints;
+        while (nextLevel < this.results.level.get() + 1 && blocks - blockAndDeathPoints < MAX_AMOUNT) {
+            nextLevel = calculateLevel(++blocks);
+        }
+        this.results.pointsToNextLevel.set(blocks - blockAndDeathPoints);
 
-	// Report
-	results.report = getReport();
-	// Set the duration
-	addon.getPipeliner().setTime(System.currentTimeMillis() - duration);
-	// All done.
+        // Report
+        results.report = getReport();
+        // Set the duration
+        addon.getPipeliner().setTime(System.currentTimeMillis() - duration);
+        // All done.
     }
 
     /**
      * @return the zeroIsland
      */
     boolean isNotZeroIsland() {
-	return !zeroIsland;
+        return !zeroIsland;
     }
 
     public void scanIsland(Pipeliner pipeliner) {
-	// Scan the next chunk
-	scanNextChunk().thenAccept(result -> {
-	    if (!Bukkit.isPrimaryThread()) {
-		addon.getPlugin().logError("scanChunk not on Primary Thread!");
-	    }
-	    // Timeout check
-	    if (System.currentTimeMillis()
-		    - pipeliner.getInProcessQueue().get(this) > addon.getSettings().getCalculationTimeout() * 60000) {
-		// Done
-		pipeliner.getInProcessQueue().remove(this);
-		getR().complete(new Results(Result.TIMEOUT));
-		addon.logError("Level calculation timed out after " + addon.getSettings().getCalculationTimeout()
-			+ "m for island: " + getIsland());
-		if (!isNotZeroIsland()) {
-		    addon.logError("Island level was being zeroed.");
-		}
-		return;
-	    }
-	    if (Boolean.TRUE.equals(result) && !pipeliner.getTask().isCancelled()) {
-		// scanNextChunk returns true if there are more chunks to scan
-		scanIsland(pipeliner);
-	    } else {
-		// Done
-		pipeliner.getInProcessQueue().remove(this);
-		// Chunk finished
-		// This was the last chunk
-		handleStackedBlocks();
-		handleChests();
-		long checkTime = System.currentTimeMillis();
-		finishTask = Bukkit.getScheduler().runTaskTimer(addon.getPlugin(), () -> {
-		    // Check every half second if all the chests and stacks have been cleared
-		    if ((chestBlocks.isEmpty() && stackedBlocks.isEmpty())
-			    || System.currentTimeMillis() - checkTime > MAX_AMOUNT) {
-			this.tidyUp();
-			this.getR().complete(getResults());
-			finishTask.cancel();
-		    }
-		}, 0, 10L);
+        // Scan the next chunk
+        scanNextChunk().thenAccept(result -> {
+            if (!Bukkit.isPrimaryThread()) {
+                addon.getPlugin().logError("scanChunk not on Primary Thread!");
+            }
+            // Timeout check
+            if (System.currentTimeMillis()
+                    - pipeliner.getInProcessQueue().get(this) > addon.getSettings().getCalculationTimeout() * 60000) {
+                // Done
+                pipeliner.getInProcessQueue().remove(this);
+                getR().complete(new Results(Result.TIMEOUT));
+                addon.logError("Level calculation timed out after " + addon.getSettings().getCalculationTimeout()
+                        + "m for island: " + getIsland());
+                if (!isNotZeroIsland()) {
+                    addon.logError("Island level was being zeroed.");
+                }
+                return;
+            }
+            if (Boolean.TRUE.equals(result) && !pipeliner.getTask().isCancelled()) {
+                // scanNextChunk returns true if there are more chunks to scan
+                scanIsland(pipeliner);
+            } else {
+                // Done
+                pipeliner.getInProcessQueue().remove(this);
+                // Chunk finished
+                // This was the last chunk
+                handleStackedBlocks();
+                handleChests();
+                long checkTime = System.currentTimeMillis();
+                finishTask = Bukkit.getScheduler().runTaskTimer(addon.getPlugin(), () -> {
+                    // Check every half second if all the chests and stacks have been cleared
+                    if ((chestBlocks.isEmpty() && stackedBlocks.isEmpty())
+                            || System.currentTimeMillis() - checkTime > MAX_AMOUNT) {
+                        this.tidyUp();
+                        this.getR().complete(getResults());
+                        finishTask.cancel();
+                    }
+                }, 0, 10L);
 
-	    }
-	});
+            }
+        });
     }
 
     private void handleChests() {
-	Iterator<Chunk> it = chestBlocks.iterator();
-	while (it.hasNext()) {
-	    Chunk v = it.next();
-	    Util.getChunkAtAsync(v.getWorld(), v.getX(), v.getZ()).thenAccept(c -> {
-		scanChests(c);
-		it.remove();
-	    });
-	}
+        Iterator<Chunk> it = chestBlocks.iterator();
+        while (it.hasNext()) {
+            Chunk v = it.next();
+            Util.getChunkAtAsync(v.getWorld(), v.getX(), v.getZ()).thenAccept(c -> {
+                scanChests(c);
+                it.remove();
+            });
+        }
     }
 
     private void handleStackedBlocks() {
-	// Deal with any stacked blocks
-	Iterator<Location> it = stackedBlocks.iterator();
-	while (it.hasNext()) {
-	    Location v = it.next();
-	    Util.getChunkAtAsync(v).thenAccept(c -> {
-		Block stackedBlock = v.getBlock();
-		boolean belowSeaLevel = seaHeight > 0 && v.getBlockY() <= seaHeight;
-		if (WildStackerAPI.getWildStacker().getSystemManager().isStackedBarrel(stackedBlock)) {
-		    StackedBarrel barrel = WildStackerAPI.getStackedBarrel(stackedBlock);
-		    int barrelAmt = WildStackerAPI.getBarrelAmount(stackedBlock);
-		    for (int _x = 0; _x < barrelAmt; _x++) {
-			checkBlock(barrel.getType(), belowSeaLevel);
-		    }
-		} else if (WildStackerAPI.getWildStacker().getSystemManager().isStackedSpawner(stackedBlock)) {
-		    int spawnerAmt = WildStackerAPI.getSpawnersAmount((CreatureSpawner) stackedBlock.getState());
-		    for (int _x = 0; _x < spawnerAmt; _x++) {
-			checkBlock(stackedBlock.getType(), belowSeaLevel);
-		    }
-		}
-		it.remove();
-	    });
-	}
+        // Deal with any stacked blocks
+        Iterator<Location> it = stackedBlocks.iterator();
+        while (it.hasNext()) {
+            Location v = it.next();
+            Util.getChunkAtAsync(v).thenAccept(c -> {
+                Block stackedBlock = v.getBlock();
+                boolean belowSeaLevel = seaHeight > 0 && v.getBlockY() <= seaHeight;
+                if (WildStackerAPI.getWildStacker().getSystemManager().isStackedBarrel(stackedBlock)) {
+                    StackedBarrel barrel = WildStackerAPI.getStackedBarrel(stackedBlock);
+                    int barrelAmt = WildStackerAPI.getBarrelAmount(stackedBlock);
+                    for (int _x = 0; _x < barrelAmt; _x++) {
+                        checkBlock(barrel.getType(), belowSeaLevel);
+                    }
+                } else if (WildStackerAPI.getWildStacker().getSystemManager().isStackedSpawner(stackedBlock)) {
+                    int spawnerAmt = WildStackerAPI.getSpawnersAmount((CreatureSpawner) stackedBlock.getState());
+                    for (int _x = 0; _x < spawnerAmt; _x++) {
+                        checkBlock(stackedBlock.getType(), belowSeaLevel);
+                    }
+                }
+                it.remove();
+            });
+        }
     }
 }

--- a/src/main/java/world/bentobox/level/calculators/UltimateStackerCalc.java
+++ b/src/main/java/world/bentobox/level/calculators/UltimateStackerCalc.java
@@ -1,0 +1,29 @@
+package world.bentobox.level.calculators;
+
+import org.bukkit.Location;
+import org.bukkit.Material;
+
+import com.craftaro.ultimatestacker.api.UltimateStackerApi;
+import com.craftaro.ultimatestacker.api.utils.Stackable;
+
+import world.bentobox.bentobox.BentoBox;
+
+/**
+ * Isolates UltimateStacker imports so that they are only loaded if the plugin exists
+ */
+public class UltimateStackerCalc {
+    public static void addStackers(Material material, Location location, Results results, boolean belowSeaLevel,
+            int value) {
+        Stackable stack = UltimateStackerApi.getBlockStackManager().getBlock(location);
+        if (stack != null) {
+            if (belowSeaLevel) {
+                results.underWaterBlockCount.addAndGet((long) stack.getAmount() * value);
+                results.uwCount.add(material);
+            } else {
+                results.rawBlockCount.addAndGet((long) stack.getAmount() * value);
+                results.mdCount.add(material);
+            }
+        }
+    }
+}
+

